app/app.py

```diff
@@ -108,7 +143,10 @@ def handle_event(self, message: str):
         media_id = event.media_id
 
         # Get the main fragment
-        fragment = self._get_fragment(filename)
+        fragment = self._get_fragment("s3_object_key", filename, 1)
```

Comment:
It's possible to have multiple essenceLinkedEvents for an essence. So this check will fail for every event except the first. It's possible to pass "IsFragment:0" to MH to check if the essence is unique.